packages/bento-frontend/src/components/Layout/LayoutView.js

```diff
@@ -28,33 +28,51 @@ import SysInfoView from '../../pages/sysInfo/view';
 import ProfileController from '../../pages/profile/profileController';
 import editUserController from '../../pages/admin/userDetails/editUserController';
 import viewUserController from '../../pages/admin/userDetails/viewUserController';
-import InActivityDialog from '../InActivityDialog';
 import OverlayWindow from '../OverlayWindow/OverlayWindow';
 import AUTH_MIDDLEWARE_CONFIG from '../Auth/authMiddlewareConfig';
 import CarView from '../../pages/cart/cartController';
+import env from '../../utils/env';
+import { PING_INTERVAL, REDIRECT_AFTER_SIGN_OUT, SHOW_WARNING_BEFORE } from '../../bento/siteWideConfig';
 
-import { AuthenticationMiddlewareGenerator } from '@bento-core/authentication';
+import { AuthenticationMiddlewareGenerator, useAuth } from '@bento-core/authentication';
+import { SessionTimeoutGenerator } from '@bento-core/session-timeout';
 
 import Notifactions from '../Notifications/NotifactionView';
 import DashTemplate from '../../pages/dashTemplate/DashTemplateController';
+import { useGlobal } from '../Global/GlobalProvider';
 
 const ScrollToTop = () => {
   window.scrollTo(0, 0);
   return null;
 };
 
-const Layout = ({ classes, isSidebarOpened }) => {
+const { SessionTimeout } = SessionTimeoutGenerator({
```

Comment:
@amattu2,
I was thinking, can we have a separate controller for the SessionTimeout component and move all the logic related to this component to that controller? This will help us to reduce complexity in the layout file.